server/raft.go

```diff
@@ -2836,6 +2856,13 @@ func (n *raft) runAsCandidate() {
 
 // handleAppendEntry handles an append entry from the wire.
 func (n *raft) handleAppendEntry(sub *subscription, c *client, _ *Account, subject, reply string, msg []byte) {
+	// If we are overwhelmed, i.e. the upper layer is not applying entries
+	// fast enough and our apply queue is building up, start to drop new
+	// append entries instead.
+	if n.Overloaded() {
```

Comment:
Do not want to acquire lock on network packets, we worked hard to try to get rid of that as much as possible. So let's consider atomic.Bool here that is set once we determine the state elsewhere.